### PR TITLE
Added a config parameter `auto.create.tables.replicated`

### PR DESCRIPTION
**Pull request by IlyaTsoi**
Solution for #62. Added a config parameter `auto.create.tables.replicated` that controls replicated tables creation.
In this logic I use a separate config parameter `use.replacingmergetree.is_deleted.column` as a flag to manage tables creation. If `true` - use `is_deleted` column if supported, otherwise use `_sign`.

### CODE REVIEW

File: sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/db/operations/ClickHouseAutoCreateTable.java [hunk 1]
```diff
@@ -75,12 +90,42 @@ public java.lang.String createTableSyntax(ArrayList<String> primaryKey, String t
         //createTableSyntax.deleteCharAt(createTableSyntax.lastIndexOf(","));
 
         // Append sign and version columns
-        createTableSyntax.append("`").append(SIGN_COLUMN).append("` ").append(SIGN_COLUMN_DATA_TYPE).append(",");
+        String DeleteColumnNameFinal = this.userDefinedDeleteColumnName;
```
Comment (subkanthi) [inline sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/db/operations/ClickHouseAutoCreateTable.java]: please use camel case for variables.

Comment (IlyaTsoi) [reply sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/db/operations/ClickHouseAutoCreateTable.java:93]: It's existant constants, not variabels.

Comment (subkanthi) [reply sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/db/operations/ClickHouseAutoCreateTable.java]: DeleteColumnNameFinal is not a constant, and it should not start with upper case.

File: sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/db/operations/ClickHouseAutoCreateTable.java [hunk 2]
```diff
@@ -75,12 +90,42 @@ public java.lang.String createTableSyntax(ArrayList<String> primaryKey, String t
...
+            }
+            DeleteColumnTypeFinal = SIGN_COLUMN_DATA_TYPE;
+        }
+        createTableSyntax.append("`").append(DeleteColumnNameFinal).append("` ").append(DeleteColumnTypeFinal).append(", ");
         createTableSyntax.append("`").append(VERSION_COLUMN).append("` ").append(VERSION_COLUMN_DATA_TYPE);
-
         createTableSyntax.append(")");
         createTableSyntax.append(" ");
-        createTableSyntax.append("ENGINE = ReplacingMergeTree(").append(VERSION_COLUMN).append(")");
+        createTableSyntax.append("ENGINE = ");
```
Comment (subkanthi) [inline sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/db/operations/ClickHouseAutoCreateTable.java:116]: Please update unit tests in ClickHouseAutoCreateTableTest.

File: sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/db/operations/ClickHouseAutoCreateTable.java (hunk 1 above)
Comment (IlyaTsoi) [reply sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/db/operations/ClickHouseAutoCreateTable.java]: OK, my bad, sorry)

**Comment by IlyaTsoi** [comment]
In my environment I use `Replicated` engine for a autocreated database if `auto.create.tables.replicated` is `true`. It's the easiest way to replicate DDL without adding `ON CLUSTER` clause in queries. I can add this feature by PR too. @subkanthi, what do you think about it?

**Comment by subkanthi** [comment]
This feature was implemented as part of other PRs , closing this for now.

**Pull request closed without merging**